### ISSUE TITLE
Scrollbars

### ISSUE THREAD

**Issue by igor-ramazanov** [sxyazi/yazi#1933]
### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (Nixpkgs 2024-09-04)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (Nixpkgs 2024-09-04)

Emulator
    Emulator.via_env: ("alacritty", "")
    Emulator.via_csi: Ok(Unknown([Sixel]))
    Emulator.detect : Unknown([Sixel])

Adapter
    Adapter.matches: Sixel

Desktop
    XDG_SESSION_TYPE           : Some("wayland")
    WAYLAND_DISPLAY            : Some("wayland-1")
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: Some("a425fbebe4cf4238e48a42f724ef2208959d66cf_1732108974_235540273")
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/run/current-system/sw/bin/zsh")
    EDITOR             : Some("kak")
    VISUAL             : Some("kak")
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : No such file or directory (os error 2)
    ZELLIJ_SESSION_NAME: Some("claid")
    Zellij version     : 0.40.1

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.2
    magick           : 7.1.1-39
    fzf              : 0.56.0
    fd               : 10.2.0
    rg               : 14.1.1
    chafa            : 1.14.4
    zoxide           : 0.9.6
    7z               : No such file or directory (os error 2)
    7zz              : 24.08
    jq               : 1.7.1


--------------------------------------------------
When reporting a bug, please also upload the `yazi.log` log file - only upload the most recent content by time.
You can find it in the "/home/igor/.local/state/yazi" directory.
```

### Please describe the problem you're trying to solve

Add a non-interactive scrollbar at the preview tab to understand the current position.

Example `lazygit`:
![image](https://github.com/user-attachments/assets/80b8accd-783e-4898-aa02-71efd5a33821)

### Would you be willing to contribute this feature?

- [ ] Yes, I'll give it a shot

### Describe the solution you'd like

Do not show scrollbars where a content fits in.

Example:
![image](https://github.com/user-attachments/assets/fa186677-f1e3-4396-95a0-517289ddb650)

### Additional context

There's a similar solution [at the `confirm` component](https://github.com/sxyazi/yazi/blob/428c92270385c55ffc2862947f2377f4247268e4/yazi-fm/src/confirm/list.rs#L37-L41).

### Validations

- [X] I have searched the existing issues/discussions
- [X] The [latest nightly build](https://yazi-rs.github.io/docs/installation/#official-binaries) doesn't already have this feature

**Comment by sxyazi** [comment]
Sorry, this is a feature that was intentionally left out during the design phase. To ensure maximum preview performance, Yazi only reads a small number of lines from the top of the file, limited to the height of the preview pane (I explained the rationale for this in https://github.com/sxyazi/yazi/issues/143). As a result, Yazi doesn't know the total number of lines in a file. Determining the total line count would require a full read of the file, no matter what.

When designing Yazi, I assumed a specific use case: the preview is mainly for quickly checking for a file, not for viewing the entire file. For complete content, a dedicated editor is better suited. This differs from Lazygit, where the purpose involves interacting with the entire file content (e.g., staging or unstaging), making a scrollbar essential to indicate the current position.

That said, I do plan to introduce a new `ui.Scrollbar` API in the future. This will allow users who don't have strict performance requirements to implement their own scrollbar in a custom previewer, in a straightforward way. At that point, `ya.preview_code()` would need to return a `ui.Text` to Lua, enabling users to combine the scrollbar with highlighted code for preview, rather than rendering it as is. Closing as not planned
